core/modules/query/QueryContext.cc

```diff
@@ -52,35 +155,65 @@ QueryContext::resolve(std::shared_ptr<ColumnRef> cr) {
             if (concrete.db.empty()) {
                 concrete.db = defaultDb;
             }
-            return concrete;
+            dbTableSet.insert(concrete);
+            return dbTableSet;
         }
     }
     // Set default db and table.
     DbTablePair p;
-    if (cr->table.empty()) { // No db or table: choose first resolver pair
-        p = resolverTables[0];
-        // TODO: We can be fancy and check the column name against the
-        // schema for the entries on the resolverTables, and choose
-        // the matching entry.
+    if (cr->table.empty()) { // No db or table.
+        if (columnToTablesMap.empty()) {
+            // This should only be the case if no connection to the database could be made, which
+            // should only happen in unit tests. This should get through the unit tests and have
```

Comment:
If the connection to the database fails, this should be the same database that the czar is using to test the user queries before sending them to workers, so the user query should fail and perhaps the czar should be terminated at that point if it can't connect. The worst case scenario here is that a query that only needs to go to one worker is sent to all workers. I think this is a slightly better option than eliminating the unit tests. Other options, such as  creating artificial  columnToTablesMap for the unit tests, is probably worthy of another ticket.